src/GapicClientTrait.php

```diff
@@ -629,6 +629,13 @@ private function createCallStack(array $callConstructionOptions)
                 'X-Goog-User-Project' => [$quotaProject]
             ];
         }
+        
+        if (isset($this->apiVersion)) {
```

Comment:
Should we also check !empty?

Reply:
Just a reminder that `isset` will return `false` when `$apiVersion` is not defined, not initialized, and when it's equal to `null`. It will return `true` when `$apiVersion` is `false` or an empty string. `!empty` will return `false` in all of these cases.

The only valid difference I could see would be when the proto defines `google.api.api_version` as an empty string:

```
google.api.api_version = ""
```

 In this case, would we want to skip it or include it as an empty header? If the former, then what we have here is correct. If the latter, then we'd want to use `!empty`.

Reply:
Hmmm I can't think of a case where the `apiVersion` is either a `null`. It can be an empty string if you write an empty api version on the proto I guess?

If so, I still think there is value in sending the value itself even if is empty. Imagine someone forgets adding an API version and then they test the client. I think seeing the empty value on the header is more valuable than not sending it at all. But is just my train of thought.

Perhaps you have a specific case in mind? I am still not an expert on the generation side of things 😄

Reply:
> Just a reminder that `isset` will return `false` when `$apiVersion` is not defined, not initialized, and when it's equal to `null`. It will return `true` when `$apiVersion` is `false` or an empty string. `!empty` will return `false` in all of these cases.
> 
> The only valid difference I could see would be when the proto defines `google.api.api_version` as an empty string:
> 
> ```
> google.api.api_version = ""
> ```
> 
> In this case, would we want to skip it or include it as an empty header? If the former, then what we have here is correct. If the latter, then we'd want to use `!empty`.

I wrote an answer before you pressed "comment" haha.

So yeah in my previous comment, I added a point where I see value in sending an empty string. That empty string is not a correct version, but I think there is value in sending whatever is in there for visibility? Don't think that obfuscating that mistake would be helpful.

Reply:
I agree, I think `isset` in here works as well or better than `empty` 👍